resolwe/flow/managers/listener.py

```diff
@@ -460,7 +462,17 @@ def handle_finish(self, obj):
                     async_to_sync(self._send_reply)(obj, {ExecutorProtocol.RESULT: ExecutorProtocol.RESULT_ERROR})
                     return
 
-                if process_rc == 0 and not d.status == Data.STATUS_ERROR:
+                if spawning_failed:
+                    process_error = copy.copy(d.process_error)
+                    process_error.append("Error while preparing spawned Data objects")
```

Comment:
Errors are appended, so there is no need to pass existing ones: https://github.com/genialis/resolwe/blob/3a1ae8077e27b5d640e3c69b64b27a4101be219b/resolwe/flow/managers/listener.py#L282